packages/components/src/font-size-picker/test/index.tsx

```diff
@@ -1,290 +1,430 @@
+		commonTests( fontSizes );
+	} );
+
+	function commonTests( fontSizes: FontSize[] ) {
+		it( 'allows custom values by default', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
 			} );
-			it( 'should fallback to font size `slug` for header hint label if `name` is undefined', () => {
-				const fontSizes = [
-					{
-						slug: 'gigantosaurus',
-						size: '1000px',
-					},
-				];
-				render(
-					<FontSizePicker
-						fontSizes={ fontSizes }
-						value={ fontSizes[ 0 ].size }
-						__nextHasNoMarginBottom
-					/>
-				);
+			const onChange = jest.fn();
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+					onChange={ onChange }
+				/>
+			);
+			await user.click(
+				screen.getByRole( 'button', { name: 'Set custom size' } )
+			);
+			await user.type( screen.getByLabelText( 'Custom' ), '80' );
+			expect( onChange ).toHaveBeenCalledWith( '80px' );
+		} );
 
-				const giganticFontSizeElement = screen.getByLabelText(
-					'Size gigantosaurus(px)'
-				);
-				expect( giganticFontSizeElement ).toBeInTheDocument();
+		it( 'does not allow custom values when disableCustomFontSizes is set', () => {
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+					disableCustomFontSizes
+				/>
+			);
+			expect(
+				screen.queryByRole( 'button', { name: 'Set custom size' } )
+			).not.toBeInTheDocument();
+		} );
+
+		it( 'does not display a slider by default', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
 			} );
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+				/>
+			);
+			await user.click(
+				screen.getByRole( 'button', { name: 'Set custom size' } )
+			);
+			// TODO: There are two spinbuttons with similar names, so can't
+			// select the slider's input in an accessible way.
+			const sliderInput = document.querySelector(
+				'.components-range-control__slider'
+			);
+			expect( sliderInput ).not.toBeInTheDocument();
 		} );
-	} );
+
+		it( 'allows a slider to be used when withSlider is set', async () => {
+			const user = userEvent.setup( {
+				advanceTimers: jest.advanceTimersByTime,
+			} );
+			const onChange = jest.fn();
+			render(
+				<FontSizePicker
+					__nextHasNoMarginBottom
+					fontSizes={ fontSizes }
+					withSlider
+					onChange={ onChange }
+				/>
+			);
+			await user.click(
+				screen.getByRole( 'button', { name: 'Set custom size' } )
+			);
+			// TODO: There are two spinbuttons with similar names, so can't
+			// select the slider's input in an accessible way.
+			const sliderInput = document.querySelector(
+				'.components-range-control__slider'
+			);
+			// Fake the slider drag using fireEvent. Testing drag-and-drop using
+			// userEvent is a job for the RangeControl tests.
+			fireEvent.change( sliderInput!, {
+				target: { value: 80 },
+			} );
```

Comment:
Unfortunately React Testing Library doesn't offer a way to test with drag events yet, so it's probably OK do leave this as-is.

I'd disagree on the "is a job for RangeControl tests" though — if there was a better way to test slider inputs, I'd argue that it should be used across all of our tests (and not just `RangeControl`). This type of consistency is great and it allows the "good patterns" to spread and be adopted around the codebase!

Reply:
Fair enough! I'll remove my editorialising comment 😛